src/Access/IAccessStorage.cpp

```diff
@@ -525,15 +525,24 @@ std::optional<AuthResult> IAccessStorage::authenticateImpl(
             if (!isAddressAllowed(*user, address))
                 throwAddressNotAllowed(address);
 
-            auto auth_type = user->auth_data.getType();
-            if (((auth_type == AuthenticationType::NO_PASSWORD) && !allow_no_password) ||
-                ((auth_type == AuthenticationType::PLAINTEXT_PASSWORD) && !allow_plaintext_password))
-                throwAuthenticationTypeNotAllowed(auth_type);
+            // for now, just throw exception in case a user exists with invalid auth method
+            // back in the day, it would also throw an exception. There might be a smarter alternative
+            // like a user scan during startup.
+            for (const auto & auth_method : user->authentication_methods)
+            {
+                auto auth_type = auth_method.getType();
+                if (((auth_type == AuthenticationType::NO_PASSWORD) && !allow_no_password) ||
+                    ((auth_type == AuthenticationType::PLAINTEXT_PASSWORD) && !allow_plaintext_password))
+                    throwAuthenticationTypeNotAllowed(auth_type);
```

Comment:
Minor note: if a user has a plaintext password and any other valid auth method, this will result in an error without trying a valid one.

Reply:
Yeah. I could also try to deny it only if the auth method was plaintext_password. What do you prefer?

Reply:
Ok, it's not so trivial to do it. `BasicCredentials` does not have a field to indicate wheather the provided credentials are plaintext.